src/test/java/com/sonatype/insight/brain/cli/CommandLineOptionFilterTest.groovy

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc. All rights reserved.
+ * Includes the third-party code listed at http://links.sonatype.com/products/nexus/attributions.
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+package com.sonatype.insight.brain.cli
+
+import groovy.cli.commons.CliBuilder
+
+import spock.lang.Specification
+
+class CommandLineOptionFilterTest
+    extends Specification
+{
+  def cli
+
+  def 'setup'() {
+    cli = new CliBuilder()
+    cli.with {
+      h(longOpt: 'help', '')
+      a(longOpt: 'authentication', '', args: 1)
+      i(longOpt: 'application-id', '', args: 1)
+      s(longOpt: 'server-url', '', args: 1)
+      e(longOpt: 'ignore-system-errors', '')
+      r(longOpt: 'result-file', '', args: 1)
+      t(longOpt: 'stage', '', args: 1)
+      X(longOpt: 'debug', '')
+      xc(longOpt: 'expanded-coverage', '')
+      w(longOpt: 'fail-on-policy-warnings', '')
+      p(longOpt: 'proxy', '', args: 1)
+      U(longOpt: 'proxy-user', '', args: 1)
+      _(longOpt: 'pki-authentication', '')
+    }
+  }
+
+  def 'not letting help and pki options thru'() {
+    setup:
+      String[] params = [
+          '-h',
+          '-a', 'auth',
+          '-i', 'appId',
+          '-s', 'serverUrl',
+          '-e',
+          '-t', 'develop',
+          '-X',
+          '-xc',
+          '-w',
+          '--pki-authentication',
+          '-p', 'proxyUser',
+          '-U', 'proxy:creds',
+          '-r', 'out.json',
+          'dir1', 'file1', '-h'
+      ]
+
+    when:
+      def args = CommandLineOptionFilter.filterCommandLineParams(params)
+      def options = cli.parse(args)
+
+    then:
+      !options.h
+      !options.'help'
+      !args.contains('-h')
+      !args.contains('--help')
+      args.contains('-a')
```

Comment:
I'm checking the args for the presence of the option and checking the parsed args for the actual value.